src/utils/timing.rs

```diff
@@ -0,0 +1,118 @@
+use std::time::Instant;
+use tracing::trace;
+
+// Measures the duration of an async operation and logs it using tracing.
+pub async fn measure_operation<T, F, Fut>(operation_name: &str, f: F) -> T
+where
+    F: FnOnce() -> Fut,
+    Fut: std::future::Future<Output = T>,
+{
+    let start = Instant::now();
+    let result = f().await;
+    let duration = start.elapsed();
+
+    trace!(
```

Comment:
Sometimes async/network/DB operations can last for a very long time (and get stuck if there's no timeout). In thatcase it is very valuable to also see the mmoent they have started. So I would prefer that we first log the start of the operation, and after it finishes, we log its duration.